ruby/v3_2_0/wl-build.sh

```diff
@@ -20,8 +22,10 @@ if [[ -z "$WASMLABS_SKIP_CONFIGURE" ]]; then
     logStatus "Configuring ruby..."
     ./configure \
         --host wasm32-unknown-wasi \
-        --with-ext=ripper,monitor \
+        --prefix=/wlr-rubies \
```

Comment:
I see here we're configuring a custom prefix to look for libraries. Does it provide more compability or any extra benefit? Not sure if there's a "convention" for the libraries folder like `/lib` or `/opt/ruby` 🤔

Reply:
This is the prefix for the whole ruby installation; this contains some things that we'll be able to clean up, such as `./include`, but also `./lib`, which contains the standard library with all configured extensions.

We can name it as we wish, but I did prefer to not use an already existing prefix such as `/lib` or `/opt` mainly because when we pack with `wasi-vfs` we will copy all contents of all directories, and if I only add `./lib/ruby` I was getting Ruby related errors to `/usr` being non-existant. I have to debug that issue, but in the meantime I think we are fine enough, given this is transparent to the user.

In general, you can still configure your search prefix with `$:` or `$LOAD_PATH`. An example:

```
ereslibre in 🌐 nuc-1 in webassembly-language-runtimes on  ruby-flavours [?] via 💎
❯ nix run nixpkgs#wasmtime -- build-output/ruby/v3_2_0/bin/ruby.wasm -- -e "puts $:"
/wlr-rubies/lib/ruby/site_ruby/3.2.0
/wlr-rubies/lib/ruby/site_ruby/3.2.0/wasm32-wasi
/wlr-rubies/lib/ruby/site_ruby
/wlr-rubies/lib/ruby/vendor_ruby/3.2.0
/wlr-rubies/lib/ruby/vendor_ruby/3.2.0/wasm32-wasi
/wlr-rubies/lib/ruby/vendor_ruby
/wlr-rubies/lib/ruby/3.2.0
/wlr-rubies/lib/ruby/3.2.0/wasm32-wasi
```

And these prefixes are set automatically by setting `--prefix`. However, you can still add other prefixes:

```
ereslibre in 🌐 nuc-1 in webassembly-language-runtimes on  ruby-flavours [?] via 💎
❯ nix run nixpkgs#wasmtime -- build-output/ruby/v3_2_0/bin/ruby.wasm -- -e "$:.unshift '/some/other/dir'; puts $:"
/some/other/dir
/wlr-rubies/lib/ruby/site_ruby/3.2.0
/wlr-rubies/lib/ruby/site_ruby/3.2.0/wasm32-wasi
/wlr-rubies/lib/ruby/site_ruby
/wlr-rubies/lib/ruby/vendor_ruby/3.2.0
/wlr-rubies/lib/ruby/vendor_ruby/3.2.0/wasm32-wasi
/wlr-rubies/lib/ruby/vendor_ruby
/wlr-rubies/lib/ruby/3.2.0
/wlr-rubies/lib/ruby/3.2.0/wasm32-wasi
```

Reply:
I see it's for the entire ruby installation. That's fine. If we come up with something more standard, we can always change :)